### PR TITLE
#6775 - Add Support for Commit+Push to Commit window

### LINKED ISSUE #6775: Extend Ctrl+Enter hotkey in Commit window to allow Push
## Feature description

Currently, the commit dialog box allows the hotkey Ctrl+Enter to commit.  This is not configurable using the hotkey settings manager.  I would like this extended to allow the hotkey Ctrl+Shift+Enter to commit as well as push.  This allows a very quick keyboard-centric workflow as follows:

Ctrl+Space to open commit dialog
Ctrl+S to stage all changes
-> focus is set to commit message box, so commit message is typed at this point.
Ctrl+Shift+Enter to commit and push

Currently the Commit + Push is broken into two steps, Ctrl+Enter to commit (with commit set to auto-close) and Ctrl+Shift+Up arrow to Push.

## Environment

- Git Extensions 3.1.1.6049
- Build 2f872105b4eb42421d9fdafd68e90a9e27d20d96
- Git 2.22.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.3752.0
- DPI 96dpi (no scaling)

### PR DESCRIPTION
**Pull request by lukevp**
Add support for Ctrl+Shift+Enter hotkey to do a Commit and Push for feature #6775
Add lukevp to Contributors file and sign contributors.txt

Fixes #6775

## Proposed changes
Add support for Ctrl+Shift+Enter hotkey to do a Commit and Push

## Test methodology 
- Tested in debug mode with old hotkey and new hotkey.  One functions as before, and the new hotkey functions properly and does both a commit and a push.
- Implemented with minimum changes while matching style of existing hotkey.

## Test environment(s) 
- Git Extensions 3.2.0
- Build 4884278fff6acbc4b8b091b6c0723060bd19232e (Dirty)
- Git 2.22.0.windows.1
- Microsoft Windows NT 10.0.18362.0
- .NET Framework 4.8.3752.0
- DPI 96dpi (no scaling)
----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/CommandsDialogs/FormCommit.cs [hunk 1]
```diff
@@ -2694,17 +2694,24 @@ private void ViewFileHistoryMenuItem_Click(object sender, EventArgs e)
...
+            {
+                ExecuteCommitCommand(push: true);
+                e.Handled = true;
+            }
+
             // Ctrl + Enter = Commit
-            if (e.Control && e.KeyCode == Keys.Enter)
+            else if (e.Control && e.KeyCode == Keys.Enter)
             {
                 ExecuteCommitCommand();
```
Comment (drewnoakes) [inline GitUI/CommandsDialogs/FormCommit.cs]: You could remove the first `if` block, keeping the old code and just changing this line to:

```c#
                ExecuteCommitCommand(push: e.Shift);
```

Comment (lukevp) [reply GitUI/CommandsDialogs/FormCommit.cs]: PR updated, thanks for the recommendation!

File: GitUI/CommandsDialogs/FormCommit.cs [hunk 2]
```diff
@@ -2694,17 +2694,18 @@ private void ViewFileHistoryMenuItem_Click(object sender, EventArgs e)
 
         private void Message_KeyUp(object sender, KeyEventArgs e)
         {
-            // Ctrl + Enter = Commit
             if (e.Control && e.KeyCode == Keys.Enter)
             {
-                ExecuteCommitCommand();
+                // Ctrl + Enter = Commit
+                // Ctrl + Shift + Enter = Commit and Push
+                ExecuteCommitCommand(e.Shift);
```
Comment (drewnoakes) [inline GitUI/CommandsDialogs/FormCommit.cs]: Using a named parameter here will make it easier to read.

```suggestion
                ExecuteCommitCommand(push: e.Shift);
```

Comment (lukevp) [reply GitUI/CommandsDialogs/FormCommit.cs]: Sorry, I missed that in your original review.  I'll add the named parameter.  I like the idea of the tooltip, however I'm not familiar with how GitExtensions is doing I18N.  Would it be acceptable for the tooltip to  say Ctrl + Enter and Ctrl + Shift + Enter, or do these need to be translatable strings?

Comment (lukevp) [reply GitUI/CommandsDialogs/FormCommit.cs]: image
image

Comment (drewnoakes) [reply GitUI/CommandsDialogs/FormCommit.cs]: Looks good. I guess these would be translated. `TranslationString` is your friend. It should be easy to add.

**Comment by RussKie** [comment]
Closing as stale. If you wish to pursue if further please ping us to re-open.
Thanks

**Pull request closed without merging**